### ISSUE TITLE
PublicApiAnalyzer fix-all adds duplicate entries to Unshipped when there are partial classes

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn-analyzers#3450]
#### Analyzer package

https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers

#### Package Version

v3.0.0-beta3.final

#### Diagnostic ID

RS0016 Symbol is not part of the declared API.

#### Repro steps

Invoke 'Fix all in document/project/solution' for this code:

```cs
// ⚠ RS0016 Symbol 'C' is not part of the declared API.
// ⚠ RS0016 Symbol 'implicit constructor for 'C'' is not part of the declared API.
//                   ↓
public partial class C
{
}

// ⚠ RS0016 Symbol 'C' is not part of the declared API.
// ⚠ RS0016 Symbol 'implicit constructor for 'C'' is not part of the declared API.
//                   ↓
public partial class C
{
}
```

#### Result

![image](https://user-images.githubusercontent.com/8040367/77855544-14cea280-71bf-11ea-9166-ecd87ed297c0.png)

```
C
C
C.C() -> void
C.C() -> void
```

This then causes the warning that there are duplicate symbols in the Unshipped file.

#### Expected result

![image](https://user-images.githubusercontent.com/8040367/77855568-36c82500-71bf-11ea-992e-0b9490634b62.png)

```
C
C.C() -> void
```

**Comment by RussKie** [comment]
Hitting the same in https://github.com/dotnet/winforms/, we have a hundred of partial files and I get a hundred of errors...

**Comment by mavasani** [comment]
This has been fixed.
